src/data/dataloader.cpp

```diff
@@ -272,14 +272,16 @@ std::optional<Dataset> loadCachedFile(const std::filesystem::path& path) {
     int valuesIdx = 0;
     for (Dataset::Entry& e : result.entries) {
         e.data.resize(nValues);
-        std::memcpy(e.data.data(), entriesBuffer.data() + valuesIdx, nValues);
+        auto startOfCopy = entriesBuffer.begin() + valuesIdx;
+        std::move(startOfCopy, startOfCopy + nValues, e.data.begin());
         valuesIdx += nValues;
 
         if (e.comment.has_value()) {
             ghoul_assert(commentIdx < commentBuffer.size(), "Index too large");
 
             // If we have a comment, we need to extract its length's worth of characters
             // from the buffer
+            // @TODO replace this with std::move as well?
             std::memcpy(e.comment->data(), &commentBuffer[commentIdx], e.comment->size());
```

Comment:
This just happens to work since the data being copied is a string, if I understand it correctly?  Should we go with the `std::move` approach here too?